### PR TITLE
added @TestInstance(PER_CLASS) to ContainerMatrix-Tests

### PR DESCRIPTION
**Pull request by janheise**
## Description

/jenkins-pr-deps Graylog2/graylog-plugin-enterprise#2963

## Motivation and Context
The general modus-operande for container-matrix tests is now @TestInstance(PER_CLASS) because it is a good fit. It also enables that @BeforeAll can be used in our testing-contexts because the need ceases to be static.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: full-backend-tests/src/test/java/org/graylog/plugins/views/SearchMetadataIT.java [hunk 1]
```diff
@@ -80,6 +91,7 @@ void testMinimalRequestWithSingleParameter() {
 
     @ContainerMatrixTest
     void testRetrievingMetadataForStoredSearchWithoutParameter() {
+        LOG.error("testRetrievingMetadataForStoredSearchWithoutParameter()");
```
Comment (todvora) [inline full-backend-tests/src/test/java/org/graylog/plugins/views/SearchMetadataIT.java]: Should we remove this logging before merging?

File: full-backend-tests/src/test/java/org/graylog/plugins/views/SearchMetadataIT.java [hunk 2]
```diff
@@ -93,6 +105,7 @@ void testRetrievingMetadataForStoredSearchWithoutParameter() {
 
     @ContainerMatrixTest
     void testRetrievingMetadataForStoredSearchWithParameter() {
+        LOG.error("testRetrievingMetadataForStoredSearchWithParameter()");
```
Comment (todvora) [inline full-backend-tests/src/test/java/org/graylog/plugins/views/SearchMetadataIT.java]: Should we remove this logging before merging?

**Review by todvora — CHANGES_REQUESTED** [review]
Looks good, thanks! Only remove those (forgotten) two logging lines maybe?

File: full-backend-tests/src/test/java/org/graylog/plugins/views/SearchMetadataIT.java (hunk 2 above)
Comment (janheise) [reply full-backend-tests/src/test/java/org/graylog/plugins/views/SearchMetadataIT.java]: Thanks for the review - I already removed those logging messages.

File: full-backend-tests/src/test/java/org/graylog/plugins/views/SearchMetadataIT.java (hunk 1 above)
Comment (janheise) [reply full-backend-tests/src/test/java/org/graylog/plugins/views/SearchMetadataIT.java]: I already did

**Review by todvora — APPROVED** [review]
Perfect, thanks!

**Pull request merged**